Stop hardcoding workspace name in remote state configurations

jsf9k opened the pull request:
## 🗣 Description

The workspace name in remote state configurations was previously hardcoded to "production".  This PR changes the code to instead use `terraform.workspace`, so that the workspace name changes as we switch between the production and staging environments.

## 💭 Motivation and Context

We need to be able to switch as seamlessly as possible between the production and staging environments.  The more manual foo that is required the more likely that we make a mistake and damage production.

## 🧪 Testing

These changes were successfully deployed to staging.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Stop hardcoding workspace name in remote state configurations

dav3r commented:
@jsf9k - I don't think this will work, since the workspaces in this project are `env0`, `env1`, etc.

jsf9k commented:
Something needs to be done with the remote state configurations whose `workspace` attribute is set to `var.assessment_account_name`.  How about `terraform.workspace == "production" ? var.assessment_account_name : "${var.assessment_account_name}-${terraform.workspace}"`?

Ideally we could make the existing workspace names `env0-production`, for example, but that would involve renaming some workspaces.  The TF code above assumes we are unwilling to do that.

mcdonnnj reviewed (approved):
This is the good stuff 🍻

jsf9k committed:
Fix copious errors in previous approach

@dav3r pointed out that the workspace names for this TF code are env0,
env1, etc.  Therefore my previous approach would not work.  These
changes fix that.

jsf9k commented:
OK, @dav3r, please have another look when you have time.

felddy commented on remote_states.tf:
```diff
@@ -45,7 +59,7 @@ data "terraform_remote_state" "images" {
   }
 
-  workspace = "production"
+  workspace = local.workspace_type
```
Shouldn't this be `local.workspace_name` ?
(turtles all the way down) ⬇️

jsf9k replied:
I don't think so.  `local.workspace_name` would be something like "env0-staging".  `local.workspace_type` would be "staging".  Or are you seeing something I am missing?

felddy replied:
It's all good.  I was misreading the code, and was confused at line 47.

felddy reviewed (approved):
~I think there is a few bugs.  See my comments.~
~Otherwise,~ I think this will work.

dav3r commented on remote_states.tf:
```diff
@@ -30,7 +44,7 @@ data "terraform_remote_state" "dynamic_assessment" {
   }
 
-  workspace = var.assessment_account_name
+  workspace = local.workspace_name
```
It's probably worth a brief comment here to explain why this one is different from the others (even though things are well-explained in the comment at the top of this file).

felddy replied:
This is what threw me off.  Until I expanded the whole file.

jsf9k replied:
Good idea.  Comment added in commit f9b0484.

dav3r reviewed (approved):
Yessir, this will do nicely. 👍 
Please consider my one request for an additional comment.

jsf9k committed:
Add a comment explaining why one remote state config uses a different value for workspace than all the others

Thanks to @dav3r for the suggestion.

jsf9k committed:
Merge branch 'develop' into bugfix/do-not-hardcode-workspace-in-remote-state
